completions/java

```diff
@@ -236,6 +236,9 @@ _comp_cmd_java()
         else
             # classes completion
             _comp_cmd_java__classes
+            # support for launching source-code
+            # programs (JEP 330, JEP 458)
+            _filedir 'java'
```

Comment:
`_filedir` is an old API. We now use the new API in the bash-completion codebase.

```suggestion
            _comp_compgen -a filedir 'java'
```

Reply:
`_filedir 'java'` replaced with `_comp_compgen -a filedir 'java'`.